File: Guías Completas/Aritmética Modular/Contracts/Python/Aritmetica_Modular.py
def method_1(a,b):

    for i in range(1,509):
        result = (a ** i) % b
        print("Result", i,":", result)

print("Method 1:")
print("")
method_1(499,509)

print("")

def method_2(a,b):

    for i in range(1,509):
        result = (a ** i) % b
        if result == 466:
            print("Result is:",i)

print("Method 2:")
print("")
method_2(499,509)

File: Guías Completas/Aritmética Modular/Contracts/Rust/Aritmetica_Modular.rs
fn method_1(a: u64, b: u64) {
    for i in 1..509 {
        let result = a.pow(i) % b;
        println!("Result {}: {}", i, result);
    }
}

fn method_2(a: u64, b: u64) {
    for i in 1..509 {
        let result = a.pow(i) % b;
        if result == 466 {
            println!("Result is: {}", i);
        }
    }
}

fn main() {
    println!("Method 1:\n");
    method_1(499, 509);

    println!("\nMethod 2:\n");
    method_2(499, 509);
}
